azure-pipelines.yml

```diff
@@ -0,0 +1,49 @@
+trigger:
+- main
+- features/*
+- releases/*
+
+jobs:
+#################################################
+- job: 'windows'
+#################################################
+  displayName: Build on Windows
+  pool:
+    vmImage: windows-2019
+  steps:
+  - template: ci/azure-pipelines-steps.yml
+
+#################################################
+- job: 'linux'
+#################################################
+  displayName: Build on Linux
+  pool:
+    vmImage: ubuntu-20.04
+  steps:
+  - template: ci/azure-pipelines-steps.yml
+
+  - task: PublishPipelineArtifact@1
+    inputs:
+      targetPath: '_build'
+      artifactType: 'pipeline'
+      artifactName: 'npm-package'
+    condition: and(succeeded(), eq(variables['Build.SourceBranch'], 'refs/heads/main'))
+
+  # For CI runs on master, automatically publish packages
+  - bash: |
+        echo "//registry.npmjs.org/:_authToken=${NPM_TOKEN}" > .npmrc
+        cd _build
+        npm publish || true # Ignore publish failures, usually will happen because package already exists
+    displayName: NPM Publish
+    condition: and(succeeded(), in(variables['build.reason'], 'IndividualCI', 'BatchedCI', 'Manual'), eq(variables['Build.SourceBranch'], 'refs/heads/main'))
+    env:
+      NPM_TOKEN: $(npmPublishToken)
```

Comment:
Can we make this step through [npm task](https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/reference/npm-v1?view=azure-pipelines&viewFallbackFrom=azure-devops)? This will allow us to use one npm service  connection per each npm packages in the future.
https://docs.microsoft.com/en-us/azure/devops/pipelines/artifacts/npm?view=azure-devops&tabs=yaml